src/tabmat/ext/cat_split_helpers-tmpl.cpp

```diff
@@ -1,26 +1,29 @@
 #include <vector>
 #include <omp.h>
 
-<%def name="transpose_matvec(dropfirst)">
+<%def name="transpose_matvec(type)">
 template <typename Int, typename F>
-void _transpose_matvec_${dropfirst}(
+void _transpose_matvec_${type}(
     Int n_rows,
     Int* indices,
     F* other,
     F* res,
     Int res_size
+    % if type == 'all_rows_complex':
+        , bool drop_first
+    % endif
 ) {
     int num_threads = omp_get_max_threads();
     std::vector<F> all_res(num_threads * res_size, 0.0);
-    #pragma omp parallel shared(all_res) 
+    #pragma omp parallel shared(all_res)
     {
 	int tid = omp_get_thread_num();
-	F* res_slice = &all_res[tid * res_size]; 
+	F* res_slice = &all_res[tid * res_size];
 	#pragma omp for
         for (Py_ssize_t i = 0; i < n_rows; i++) {
-            % if dropfirst == 'all_rows_drop_first':
-                Py_ssize_t col_idx = indices[i] - 1;
-                if (col_idx != -1) {
+            % if type == 'all_rows_complex':
+                Py_ssize_t col_idx = indices[i] - drop_first;
```

Comment:
```suggestion
                Py_ssize_t col_idx = indices[i] - static_cast<Int>(drop_first);
```

I don't think it really makes a difference right now, but it might be better to be explicit in case this is refactored in the future.